### DISCUSSION TITLE
Getting Microsoft.AspNetCore.OData.Versioning to work with Microsoft.AspNetCore.OData v8.0.11?

### DISCUSSION THREAD

**Discussion by jwstemper** [dotnet/aspnet-api-versioning#881] [Q&A] [answered]
The latest version of Microsoft.AspNetCore.OData.Versioning (5.0.0) specifics that it works with  Microsoft.AspNetCore.OData less than v8.  Is there something in the works to work with versions newer? i.e. 8.0.11?

Is there a preferred way to do versioning other than the Microsoft.AspNetCore.OData.Versioning library?

**Comment by commonsensesoftware** [comment, marked as answer]
Yes - and no. OData 8.x+ is supported in `6.0` which has been rebranded to **Asp.Versioning.OData**. The rationale behind all of this is not short and there are a detailed background discussions in #807 and #808. There is also a callout banner in the repo `README.md` for the updates as well as updated links to all the new packages.

Sadly, I've gotten very minimal support from Microsoft or the .NET Foundation to make this transition smoother. I'm still blocked from publishing updates to `5.x` packages with the `Microsoft.*` prefix despite being a co-owner and having the proper code signing setup. I'm not sure if this will ever be solved or in a meaningful timeframe. By the time it is solved, it may be irrelevant. I've fallen on every possible sword and grenade to fight for what's right for the community leveraging this project (e.g. _Customer Obsession_), but I have failed to be able to make smooth. I knew it would be painful or easy to miss for people, which is why I fought so hard for it. Unfortunately, without being able to publish any updates to NuGet, it's not obvious that the existing packages are spinning down in favor of new packages. Stating that Microsoft _doesn't care_ is probably too harsh, but after 18+ months of transition, it certainly hasn't been a _priority_. It was almost a year before I could get `6.0` out the door, but I might as well have created a new repo and started over (however that would have been even more confusing for the community). #end-kvetch

To fully answer your question, API Versioning `6.0` supports .NET 6.0 as well as OData `8.x`. OData batching is currently broken, but I finally have a solution for it which will come in the `6.1` release. I expect `6.1` to be ready sometime during the next week or less. I will then be gearing for .NET 7.0 support. I'm not familiar with what OData's plans or roadmap are.
